### PR TITLE
Add WireGuard (wg) Command Output Parser

### LINKED ISSUE #605: Add WireGuard (wg) Command Output Parser
## Description
I propose a new parser for the jc utility to handle `wg show all dump` command output from WireGuard. The goal is to convert this output into structured JSON.

#### Proposed Schema
```[
  {
    "device": string,
    "privateKey": string,
    "publicKey": string,
    "listenPort": integer,
    "fwmark": integer,
    "peers": [
      {
        "publicKey": string,
        "presharedKey": string,
        "endpoint": string,
        "latestHandshake": integer,
        "transferRx": integer,
        "transferSx": integer,
        "persistentKeepalive": integer,
        "allowedIps": [string]
      }
    ]
  }
]
```
#### Example Input Data
```
wg0    aEbVdvHSEp3oofHDNVCsUoaRSxk1Og8/pTLof5yF+1M=    OIxbQszw1chdO5uigAxpsl4fc/h04yMYafl72gUbakM=    51820   off
wg0    sQFGAhSdx0aC7DmTFojzBOW8Ccjv1XV5+N9FnkZu5zc=    (none)  79.134.136.199:40036    10.10.0.2/32    1728809756  1378724  406524  off
wg0    B9csmpvrv4Q7gpjc6zAbNNO8hIOYfpBqxmik2aNpwwE=    (none)  79.134.136.199:35946    10.10.0.3/32    1728809756  4884248  3544596  off
wg0    miiSYR5UdevREhlWpmnci+vv/dEGLHbNtKu7u1CuOD4=    (none)  (none)  10.10.0.4/32    0   0   0   off
wg0    gx9+JHLHJvOfBNjTmZ8KQAnThFFiZMQrX1kRaYcIYzw=    (none)  173.244.225.194:45014   10.10.0.5/32    1728809827  1363652  458252  off
```
#### Example JSON output
```
[
  {
    "device": "wg0",
    "privateKey": "aEbVdvHSEp3oofHDNVCsUoaRSxk1Og8/pTLof5yF+1M=",
    "publicKey": "OIxbQszw1chdO5uigAxpsl4fc/h04yMYafl72gUbakM=",
    "listenPort": 51820,
    "fwmark": null,
    "peers": {
      "sQFGAhSdx0aC7DmTFojzBOW8Ccjv1XV5+N9FnkZu5zc=": {
        "endpoint": "79.134.136.199:40036",
        "latestHandshake": 1728809756,
        "transferRx": 1378724,
        "transferSx": 406524,
        "persistentKeepalive": -1,
        "allowedIps": ["10.10.0.2/32"]
      },
      "B9csmpvrv4Q7gpjc6zAbNNO8hIOYfpBqxmik2aNpwwE=": {
        "endpoint": "79.134.136.199:35946",
        "latestHandshake": 1728809756,
        "transferRx": 4884248,
        "transferSx": 3544596,
        "persistentKeepalive": -1,
        "allowedIps": ["10.10.0.3/32"]
      }
    }
  }
]
```

### PR DESCRIPTION
**Pull request by Luigi31415**
### Description
This PR adds a new parser for handling the output of the `wg show all dump` command from WireGuard. The parser converts the command output into structured JSON, making it easier for users to integrate WireGuard data into their workflows for monitoring or automation.
---
### Summary of Changes
- Added a parser that handles the wg show all dump command output.
- The parser produces structured JSON with detailed device and peer information, including fields such as device, privateKey, publicKey, listenPort, peers, and more.
---
### Request for Collaboration
- Test Cases: I would appreciate help with generating comprehensive test cases to cover possible variations of the `wg` command output. This will ensure that the parser handles all edge cases effectively.
- Review Feedback: Any feedback on the schema and implementation would be very helpful for refinement.

Closes #605

### CODE REVIEW

File: jc/lib.py [hunk 1]
```diff
@@ -224,6 +224,7 @@
     'vmstat-s',
     'w',
     'wc',
+    'wg',
```
Comment (joehacksalot) [inline jc/lib.py:227]: It might be a good idea to use wg-show to account for other possible future wg command parsers (e.g. showconf)

Comment (Luigi31415) [reply jc/lib.py]: @joehacksalot Done, thanks for the heads up 🫡

File: jc/parsers/wg.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                    "publicKey": peer_key,
+                    "presharedKey": peer_data.get("presharedKey"),
+                    "endpoint": peer_data.get("endpoint"),
+                    "latestHandshake": peer_data.get("latestHandshake", 0),
+                    "transferRx": peer_data.get("transferRx", 0),
+                    "transferSx": peer_data.get("transferSx", 0),
+                    "persistentKeepalive": peer_data.get("persistentKeepalive", -1),
+                    "allowedIps": peer_data.get("allowedIps", [])
+                }
+                for peer_key, peer_data in device.get("peers", {}).items()
```
Comment (kellyjonbrazil) [inline jc/parsers/wg.py:185]: Ha! I've never seen that before. Clever!

File: tests/test_wg.py
Comment (kellyjonbrazil) [inline tests/test_wg.py:1]: probably want to change the name of this file to `test_wg_show.py`

File: tests/test_wg.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+sys.path.pop()
+
+class MyTests(unittest.TestCase):
+    def test_wg_nodata(self):
+        """
+        Test 'wg' with no data
+        """
+        test_utils.run_no_data(self, __file__, [])
+
+    def test_foo_all_fixtures(self):
```
Comment (kellyjonbrazil) [inline tests/test_wg.py:15]: change function name to `test_wg_show_all_fixtures`

File: tests/test_wg.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import unittest
+import os
+import sys
+sys.path.append(os.getcwd())
+from tests import utils_for_test as test_utils
+sys.path.pop()
+
+class MyTests(unittest.TestCase):
+    def test_wg_nodata(self):
```
Comment (kellyjonbrazil) [inline tests/test_wg.py:9]: Change function name to `test_wg_show_nodata`

**Review by kellyjonbrazil — CHANGES_REQUESTED** [review]
Thanks for the contribution!

**Comment by Luigi31415** [comment]
Hi @kellyjonbrazil,

Thank you for reviewing this PR! I wanted to note that the output of `wg` is consistent across most operating systems. Based on this, I believe we can confidently proceed with merging this PR by including tests for Ubuntu and Windows, as they should be sufficient to cover the majority of use cases.

Please let me know if there’s anything else I can assist with to help finalize this.

Best regards,
Luigi

**Review by kellyjonbrazil — APPROVED** [review]
